WcaOnRails/app/models/competition.rb

```diff
@@ -824,6 +824,25 @@ def to_wcif
     }
   end
 
+  def load_wcif!(wcif)
+    # Remove extra events.
+    self.competition_events.each do |competition_event|
+      competition_event.destroy! unless wcif["events"].find { |wcif_event| wcif_event["id"] == competition_event.event.id }
+    end
```

Comment:
Hmm... that looks nicer, but there's no `destroy_all!` method, so I worry this could silently fail to delete rows from the database. [The docs](https://apidock.com/rails/ActiveRecord/Relation/destroy_all) say this method "Returns the collection of objects that were destroyed", which I think means we'd have to check that the set of objects it returns is the same as the set of objects we were trying to destroy, which I do not think would be cleaner than the current implementation.